4_predict/src/plot_diagnostics.R

```diff
@@ -0,0 +1,559 @@
+                                          perf_metric = NULL, out_dir){
+  #' @description Plots hyperparameter optimization results
+  #'
+  #' @param opt_result output of model hyperparameter optimization
+  #' @param model_name name to append to file name
+  #' @param plt_type passed to tune::autoplot type parameter
+  #' @param perf_metric performance metric passed to tune::autoplot metric parameter
+  #' Leave as NULL to plot all computed metrics.
+  #' @param out_dir output directory
+  #'
+  #' @return filepath to resulting plot
+  
+  fileout <- file.path(out_dir, paste0('hyperparam_marginals_', 
+                                       model_name, '.png'))
+  
+  p1 <- tune::autoplot(object = opt_result, type = plt_type, 
+                       metric = perf_metric) +
+    ggtitle(model_name)
+  
+  ggsave(filename = fileout, plot = p1, device = 'png')
+  
+  return(fileout)
+}
+
+plot_vip <- function(RF_model, model_name, num_features, out_dir){
+  #' 
+  #' @description Plots the variable importance plot from a RF model
+  #'
+  #' @param RF_model workflow containing the best trained RF model, fit to all training data 
+  #' @param model_name name to append to file name
+  #' @param num_features select the top num_features number of features to plot
+  #' @param out_dir output directory
+  #' 
+  #' @return filepath to resulting plot
+  #' 
+  
+  fileout <- file.path(out_dir, paste0('vip_', model_name, '.png'))
+  
+  p1 <- vip::vip(RF_model %>% extract_fit_parsnip(), 
+            num_features = num_features, aesthetics = list(width = 0.6)) +
+    ggtitle(model_name) +
+    theme(axis.title.x = element_text(size = 18),
+          axis.text.y = element_text(size = 18))
+  
+  ggsave(filename = fileout, plot = p1, device = 'png')
+  
+  return(fileout)
+}
+
+
+plot_pred_obs <- function(df_pred_obs, model_name, out_dir,
+                          from_predict = FALSE, model_wf = NULL, pred_data = NULL,
+                          pred_var = NULL, count_shade = FALSE){
+  #' @description returns a scatterplot comparing the predicted and observed values
+  #'
+  #' @param df_pred_obs df with obs and .pred columns
+  #' @param model_name name to append to file name
+  #' @param out_dir output directory
+  #' @param from_predict logical stating if predictions should be made within
+  #' this function using the provided model_wf and pred_data
+  #' @param model_wf model workflow
+  #' @param pred_data new_data for predict.workflow
+  #' @param pred_var the column name of the variable to be predicted
+  #' @param count_shade logical indicating whether or not to make a count shaded
+  #' density scatterplot instead of a traditional scatterplot
+  #'
+  #' @return filepath to the resulting plot
+  
+  if(from_predict){
+    #predict from provided workflow and data
+    df_pred_obs <- predict(model_wf, new_data = pred_data, type = 'numeric') %>%
+      mutate(obs = pred_data[[pred_var]])
+  }
```

Comment:
Not currently. I copied most of this function from the FHWA repo where we use the `from_predict` capability instead of making a target containing the new predictions in an object like `df_pred_obs`. In this repo I've created targets with the predictions because the predictions are used for other plots and targets

Reply:
Ah, OK. I generally like the idea of having the plotting function stick to just creating plots, but I'll keep this in mind.